analyzers/src/SonarAnalyzer.CSharp/Rules/SymbolicExecution/SymbolicExecutionRunner.cs

```diff
@@ -112,24 +80,29 @@ protected override void Initialize(SonarAnalysisContext context)
                 c =>
                 {
                     var declaration = (AnonymousFunctionExpressionSyntax)c.Node;
-                    var symbol = c.SemanticModel.GetSymbolInfo(declaration).Symbol;
-                    if (symbol == null)
+                    if (c.SemanticModel.GetSymbolInfo(declaration).Symbol is { } symbol)
```

Comment:
I think it's possible to replace this with 
```
c => Analyze<AnonymousFunctionExpressionSyntax>(c, x => x.Body),
```

This will make all callbacks uniform. I have to admit I spent a bit wondering why this was special.

Reply:
No, we can't. `AnonymousFunctionExpressionSyntax` needs `GetSymbolInfo` while the rest needs `GetDeclaredSymbol`.

That's the reason why this one is a little different and `Analyze` is split in two overloads.